app/models/session/sms.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+class Session::Sms < Session
+  SMS_CODE_MIN_LENGTH = 4
+  validates :sms_code, uniqueness: true, presence: true, length: { minimum: SMS_CODE_MIN_LENGTH }, if: -> { published? }
+  validates :question_group_finish, absence: true
+
+  attribute :settings, :json, default: -> { { 'narrator' => { 'voice' => false, 'animation' => false } } }
+
+  after_commit :create_core_children, on: :create
+
+  def first_question
+    question_groups.where('questions_count > 0').order(:position).first.questions.order(:position).first
+  end
+
+  def translate_questions(translator, source_language_name_short, destination_language_name_short)
+    questions.each do |question|
+      question.translate(translator, source_language_name_short, destination_language_name_short)
+    end
+  end
+
+  def integral_update
+    return if published?
+
+    save!
+  end
```

Comment:
We've defined this method in the Session model. I believe this occurrence is redundant.